### DISCUSSION TITLE
FunctionID: library function without a name

### DISCUSSION THREAD

**Discussion by lab313ru** [NationalSecurityAgency/ghidra#5125] [Q&A]
Is it possible to specify that some function is a library one, if it has no name, but was found in fidb?

**Comment by astrelsky** [comment]
> Is it possible to specify that some function is a library one, if it has no name, but was found in fidb?

Do you mean that the function contained within the fidb itself has no name?

**Comment by lab313ru** [comment]
> > Is it possible to specify that some function is a library one, if it has no name, but was found in fidb?
> 
> Do you mean that the function contained within the fidb itself has no name?

Exactly.

**Comment by astrelsky** [comment]
> > > Is it possible to specify that some function is a library one, if it has no name, but was found in fidb?
> > 
> > 
> > Do you mean that the function contained within the fidb itself has no name?
> 
> Exactly.

Well when I function has no name the decompiler freaks out and refuses to decompile the function. This would require a function in ghidra to have no name which I don't think is allowed and is only obtainable through a bug. I can't really say for sure but considering the above I don't believe so.

**Comment by lab313ru** [comment]
Example: obj file has an exported function, and a function that is used internally only. The last one will be without a name. But I want to mark it as related to that object file (library one).
